File: app/src/main/java/com/diamond/appcliente/viewmodel/ListarReservaViewModel.java
package com.diamond.appcliente.viewmodel;

import android.app.Application;
import android.util.Log;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import com.diamond.appcliente.api.ApiClient;
import com.diamond.appcliente.api.AuthApiService;
import com.diamond.appcliente.dto.common.ApiResponse;
import com.diamond.appcliente.dto.reserva.ReservaListResponse;
import com.diamond.appcliente.dto.reserva.ReservaResponse;
import java.util.List;
import okhttp3.MultipartBody;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ListarReservaViewModel extends AndroidViewModel {
    private AuthApiService apiService;
    /* access modifiers changed from: private */
    public MutableLiveData<List<ReservaResponse>> reservasLiveData = new MutableLiveData<>();

    public interface ActualizarCallback {
        void onError(String str);

        void onSuccess(String str);
    }

    public ListarReservaViewModel(Application application) {
        super(application);
        this.apiService = (AuthApiService) ApiClient.getRetrofit(application.getApplicationContext(), true).create(AuthApiService.class);
    }

    public LiveData<List<ReservaResponse>> getReservas() {
        loadReservations();
        return this.reservasLiveData;
    }

    private void loadReservations() {
        Log.d("ListarReservaViewModel", "Haciendo llamada a la API para obtener reservas...");
        this.apiService.listarMisReservas().enqueue(new Callback<ReservaListResponse>() {
            public void onResponse(Call<ReservaListResponse> call, Response<ReservaListResponse> response) {
                Log.d("ListarReservaViewModel", "Respuesta de la API: " + response.code());
                if (!response.isSuccessful() || response.body() == null) {
                    Log.d("ListarReservaViewModel", "No se encontraron reservas o error en la respuesta.");
                    ListarReservaViewModel.this.reservasLiveData.setValue((List<ReservaResponse>) null);
                    return;
                }
                ListarReservaViewModel.this.reservasLiveData.setValue(((ReservaListResponse) response.body()).getData());
                Log.d("ListarReservaViewModel", "Reservas recibidas: " + ((ReservaListResponse) response.body()).getData().toString());
            }

            public void onFailure(Call<ReservaListResponse> call, Throwable t) {
                Log.e("ListarReservaViewModel", "Error en la llamada a la API: ", t);
                ListarReservaViewModel.this.reservasLiveData.setValue((List<ReservaResponse>) null);
            }
        });
    }

    public void subirComprobante(Long reservaId, MultipartBody.Part imagenPart, final ActualizarCallback callback) {
        this.apiService.subirComprobante(reservaId, imagenPart).enqueue(new Callback<ApiResponse<Object>>() {
            public void onResponse(Call<ApiResponse<Object>> call, Response<ApiResponse<Object>> response) {
                if (response.isSuccessful()) {
                    callback.onSuccess("Comprobante subido correctamente.");
                } else {
                    callback.onError("Error al subir el comprobante.");
                }
            }

            public void onFailure(Call<ApiResponse<Object>> call, Throwable t) {
                callback.onError(t.getMessage());
            }
        });
    }
}

File: app/src/main/java/com/diamond/appcliente/viewmodel/ListarReservaViewModel.kt
package com.diamond.appcliente.viewmodel

import android.content.Context
import android.net.Uri
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.diamond.appcliente.dto.reserva.ReservaResponse
import com.diamond.appcliente.repository.ReservaRepository
import com.shared.models.ui.state.UiState
import dagger.hilt.android.lifecycle.HiltViewModel
import dagger.hilt.android.qualifiers.ApplicationContext
import kotlinx.coroutines.flow.MutableSharedFlow
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.SharedFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asSharedFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch
import okhttp3.MediaType
import okhttp3.MultipartBody
import okhttp3.RequestBody
import java.io.File
import javax.inject.Inject

@HiltViewModel
class ListarReservaViewModel @Inject constructor(
    private val reservaRepository: ReservaRepository,
    @ApplicationContext private val context: Context
) : ViewModel() {

    private val _reservas = MutableStateFlow<UiState<List<ReservaResponse>>>(UiState.Idle)
    val reservas: StateFlow<UiState<List<ReservaResponse>>> = _reservas.asStateFlow()

    private val _recompensaCount = MutableStateFlow(0)
    val recompensaCount: StateFlow<Int> = _recompensaCount.asStateFlow()

    private val _comprobanteEvento = MutableSharedFlow<String>()
    val comprobanteEvento: SharedFlow<String> = _comprobanteEvento.asSharedFlow()

    private val _error = MutableSharedFlow<String>()
    val error: SharedFlow<String> = _error.asSharedFlow()

    fun cargarReservas() {
        viewModelScope.launch {
            _reservas.value = UiState.Loading
            try {
                val lista = reservaRepository.listarMisReservas()
                _recompensaCount.value = lista.count { it.estRecompensa == 0 }
                _reservas.value = UiState.Success(lista)
            } catch (e: Exception) {
                _reservas.value = UiState.Error(e.message ?: "Error desconocido")
            }
        }
    }

    fun subirComprobante(reservaId: Long, uri: Uri) {
        viewModelScope.launch {
            try {
                val inputStream = context.contentResolver.openInputStream(uri)!!
                val file = File(context.cacheDir, "comprobante.jpg")
                file.outputStream().use { inputStream.copyTo(it) }
                inputStream.close()
                val imagenPart = MultipartBody.Part.createFormData(
                    "imagen", file.name,
                    RequestBody.create(MediaType.parse("image/*"), file)
                )
                _comprobanteEvento.emit(reservaRepository.subirComprobante(reservaId, imagenPart))
            } catch (e: Exception) {
                _error.emit(e.message ?: "Error desconocido")
            }
        }
    }
}
